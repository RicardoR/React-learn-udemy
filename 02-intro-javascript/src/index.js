import { giphyRandomEndpoint, apiKey } from './data/giphy-data';

const getImagePromise = () => Promise.resolve('image with promise');

getImagePromise().then((image) => {
  console.log(image);
});

const getImageAsync = async () => {
  try {
    const fetchRequest = await fetch(
      `${giphyRandomEndpoint}?api_key=${apiKey}`
    );
    const { data } = await fetchRequest.json();
    return data.images.original.url;
  } catch (error) {
    console.error(error);
  }
};

getImageAsync()
  .then((imageUrl) => {
    const imageElement = document.createElement('img');
    imageElement.src = imageUrl;
    document.body.appendChild(imageElement);
  })
  .catch((error) => console.info(error));
