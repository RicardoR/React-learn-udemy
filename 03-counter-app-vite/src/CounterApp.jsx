 import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {

    const [counter, updateCounter] = useState(value);

    const updateValue = (val) => {
        updateCounter(() => counter + val )
    };

    return (
        <>
            <h1> CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={() => updateValue(1)}>+1</button>
            <button onClick={() => updateValue(2)}>+2</button>
            <button onClick={() => updateValue(-1)}>-1</button>
            <button onClick={() => updateValue(-2)}>-2</button>
        </>
    );
};


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};
