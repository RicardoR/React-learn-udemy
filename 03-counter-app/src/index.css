html, body {
    align-items: center;
    background-color: #21232a;
    color: white;
    display: flex;
    font-family: 'Roboto', sans-serif;
    font-size: 1.3rem;
    min-height: 100%;
    justify-content: center;
    flex-direction: column;
}

body > div {
    width: 100%;
    text-align: center;
}

button {
    background-color: blueviolet;
    border: none;
    border-radius: 3px;
    color: white;
    cursor: pointer;
    font-size: 0.8rem;
    padding: 0.2rem 0.3rem;
    transition: all 0.2s ease-in-out;
    margin-left: 8px;
    margin-right: 8px;
    cursor: pointer;
}

.counter-controls {
    margin: auto;
    display: flex;
    margin-bottom: 1rem;
    align-items: center;
    align-self: center;
    justify-content: center;
}