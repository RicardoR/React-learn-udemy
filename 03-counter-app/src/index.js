import React from 'react';
import ReactDOM from 'react-dom';
import RamonesApp from './RamonesApp';
import './index.css';

const rootDiv = document.querySelector('#root');
const ramones = {
  band: 'Ramones',
  origin: 'Los Angeles, California',
};
ReactDOM.render( <RamonesApp {...ramones} />, rootDiv);