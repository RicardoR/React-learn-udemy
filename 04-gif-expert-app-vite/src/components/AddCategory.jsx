import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState();

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    };

    const onFormSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 3) {
            return;
        }

        onNewCategory(inputValue.trim());
        setInputValue('');
    };

    return (
        <form onSubmit={onFormSubmit}>
            <input
                type="text"
                placeholder="Search gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    );
};

