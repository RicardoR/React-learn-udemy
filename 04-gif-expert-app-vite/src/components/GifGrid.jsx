import PropTypes from 'prop-types';
import { getGifs } from '../helpers/getGifs';


export const GifGrid = ({category}) => {

    getGifs(category);

    return (
        <>
            <h3>{category}</h3>
        </>
    );
};


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};
