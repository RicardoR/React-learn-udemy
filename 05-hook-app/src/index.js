import React from 'react';
import ReactDOM from 'react-dom';
import { CounterWithCustomHook } from './components/01-useState/CounterWithCustomHook';
import { SimpleForm } from './components/02-useEffect/SimpleForm';
import { FormWithCustomHook } from './components/02-useEffect/FormWithCustomHook';
import { MultipleCustomHooks } from './components/03-examples/MultipleCustomHooks';
import { RealExample } from './components/04-useRef/RealExample';
import { LayoutEffect } from './components/05-useLayoutEffect/LayoutEffect';
import { Memorize } from './components/06-memos/Memorize';
import { MemoHook } from './components/06-memos/MemoHook';
import { CallbackHook } from './components/06-memos/CallbackHook';
import { Padre } from './components/07-tarea-memo/Padre';
import { TodoApp } from './components/08-useReducer/TodoApp';
import { AppRouter } from './components/09-useContext/AppRouter';
import { BrowserRouter } from 'react-router-dom';

ReactDOM.render(<CounterWithCustomHook />, document.getElementById('root'));
ReactDOM.render(<SimpleForm />, document.getElementById('simple-form'));
ReactDOM.render(
  <FormWithCustomHook />,
  document.getElementById('form-with-custom-hook')
);
ReactDOM.render(
  <MultipleCustomHooks />,
  document.getElementById('multiple-custom-hooks')
);
ReactDOM.render(<RealExample />, document.getElementById('focus-screen-hooks'));
ReactDOM.render(
  <LayoutEffect />,
  document.getElementById('layout-effect-hooks')
);
ReactDOM.render(<Memorize />, document.getElementById('memorize-hooks'));
ReactDOM.render(<MemoHook />, document.getElementById('memo-hooks'));
ReactDOM.render(<CallbackHook />, document.getElementById('callback-hooks'));
ReactDOM.render(<Padre />, document.getElementById('task-hooks'));

ReactDOM.render(<TodoApp />, document.getElementById('use-reducer'));
ReactDOM.render(
  <BrowserRouter>
    <AppRouter />
  </BrowserRouter>,
  document.getElementById('use-context')
);
