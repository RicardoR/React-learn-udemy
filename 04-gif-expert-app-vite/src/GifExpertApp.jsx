import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = () => {
        setCategories(categoryList => ['New Category', ...categoryList])
    };

    return (
        <>
            <h1> GifExpertApp</h1>
            <button onClick={onAddCategory}>Add category</button>
            <AddCategory/>
            <ol>
                {
                    categories.map((category, index) => <li key={index}>{category}</li>)
                }
            </ol>
        </>
    );
};

