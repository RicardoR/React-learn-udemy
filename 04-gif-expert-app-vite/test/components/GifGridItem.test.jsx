import {render, screen} from "@testing-library/react";
import {GifGridItem} from "../../src/components";

describe('GifGridItem UT', () => {

    test('should do match with snapshot', () => {
        const {container} = render(<GifGridItem url='image-url' title='title-image'/>);
        expect(container).toMatchSnapshot();
   });

    test('should display the image with url and alt', () => {
        render(<GifGridItem url='image-url' title='title-image'/>);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe('http://localhost/image-url');
        expect(alt).toBe('title-image');

        // expect(screen.getByRole('img').alt).toBe('title-image');
    });

    test('should display the component title', () => {
        render(<GifGridItem url='image-url' title='title-image'/>);
        expect(screen.getByText('title-image')).toBeTruthy();
    });
});
