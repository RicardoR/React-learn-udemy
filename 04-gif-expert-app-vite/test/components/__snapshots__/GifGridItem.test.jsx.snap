// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem UT should do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title-image"
      src="image-url"
    />
    <p>
       
      title-image
    </p>
  </div>
</div>
`;
